### ISSUE TITLE
Are SEGVs while encountering particular types of monsters a known problem?

### ISSUE THREAD

**Issue by oddhack** [SLASHEM-Extended/SLASHEM-Extended#19]
I'm running from commit 184bfd4b165a360215efd06795a9492ea8b09c9a built on Debian AMD64, and finding a fairly frequent problem where encountering some specific types of monster in an adjacent cell (e.g. trying to attack them, or they attack me) will cause a SEGV. I don't see this on the list of known issues on the nethackwiki pages, but I don't know if that's up to date, either. I haven't built a debug version - not even sure how to go about that - so that's all the diagnostics I have.

Is this a known issue? Are there particular commits that are more stable I should be building from?

**Comment by AmyBSOD** [comment]
Sounds like a problem with the random stats monsters, or at least that would be my guess, I cannot really tell without knowing what the monsters are (when reporting bugs, please try to be as specific as possible, in this case mentioning the monster glyph and their color would probably help). If that is indeed the cause for this error, compiling it with the STUPIDRANDOMMONSTERBUG flag enabled (it's in config.h and disabled by default) might make the problem go away, but it will only fix future games; ones that have already been started will be screwed beyond repair, I'm afraid. :( Again, that's only if it is the bug I think it is.

**Comment by oddhack** [comment]
One monster causing this was a "nimble turret" - is that a "random stats monster"? I'm using the tty interface on Debian - whatever happens by default in an xterm - and it was the usual text color.

**Comment by AmyBSOD** [comment]
No, but it's a monster that has poison attacks, which also cause the aforementioned bug. Did the crash happen when it said "The nimble turret's attack was poisoned!"? In any case, my advice would be to recompile the game with the STUPIDRANDOMMONSTERBUG enabled in config.h, then seeing whether the problem persists.

**Comment by AmyBSOD** [comment]
Made some changes to the logic that saves those monsters. My hope is that this bug has been fixed by that change but I don't have a system that lets me reproduce it.
